network/p2p/p2p.go

```diff
@@ -164,9 +167,53 @@ func (n *p2pNetwork) Start(logger *zap.Logger) error {
 		return nil
 	}
 
-	logger.Info("starting")
+	connector := make(chan peer.AddrInfo, connectorQueueSize)
+	go func() {
+		// Wait for own subnets to be subscribed to and updated.
+		// TODO: wait more intelligently with a channel.
+		time.Sleep(8 * time.Second)
+
+		ctx, cancel := context.WithCancel(n.ctx)
+		defer cancel()
+		n.backoffConnector.Connect(ctx, connector)
+	}()
+
+	// Connect to trusted peers.
+	trustedPeers := map[peer.ID][]ma.Multiaddr{}
+	if len(n.cfg.TrustedPeers) > 0 {
+		for _, mas := range n.cfg.TrustedPeers {
+			for _, ma := range strings.Split(mas, ",") {
+				addrInfo, err := peer.AddrInfoFromString(ma)
+				if err != nil {
+					return fmt.Errorf("could not parse trusted peer: %w", err)
+				}
+				trustedPeers[addrInfo.ID] = append(trustedPeers[addrInfo.ID], addrInfo.Addrs...)
+			}
+		}
+		go func() {
+			for id, addrs := range trustedPeers {
+				connector <- peer.AddrInfo{ID: id, Addrs: addrs}
+			}
+		}()
```

Comment:
this is like extra safety here to do go routine on a buffered channel..

Reply:
@y0sher i think the code should not assume that the buffer size is higher than the number of trusted peers

Reply:
on second thought, we could simply set the buffer size to `min(256, len(trustedPeers))`, but not sure that'd be worth another round of testing 😅